### ISSUE TITLE
DeepSeek OCR: ~0.94 whole-model PCC from cumulative error in DeepseekV2 decoder stack

### ISSUE THREAD

**Issue by ashokkumarkannan1** [tenstorrent/tt-xla#4849]
### Describe the bug

- `deepseek/deepseek_ocr/pytorch-Ocr` CPU vs TT: whole-model PCC ≈ 0.94 (below 0.99). 
- Early-stop bisection shows vision + inputs_embeds are clean (PCC ≈ 1.0); 
- the drop appears in the 12× DeepseekV2DecoderLayer loop + final RMS norm. 
- Isolated run_op_test on that loop reproduces the same PCC as the full model on real doc.png inputs (≈0.951 decoder+norm, ≈0.944 with lm_head). 
- No single layer fails alone at threshold when run in isolation; PCC falls when many decoder layers run in sequence on real activations (random same-shape inputs pass ≈0.999).

#### Call chain (where it shows up)
```
DeepseekOCRForCausalLM
  → DeepseekOCRModel
      → SAM ViT + CLIP-L + projector + masked_scatter into inputs_embeds   # PCC ≈ 1.0 (bisection)
      → DeepseekV2Model.forward
          → 12 × DeepseekV2DecoderLayer (layer 0 dense; layers 1–11 MoE)
          → DeepseekV2RMSNorm
  → lm_head   # small extra drop (~0.951 → ~0.944)
```
- Drop is isolated before lm_head on real OCR inputs: cumulative error in the decoder stack on device vs CPU golden, not in the vision prefix alone.

Isolated decoder-stack sanity (run_op_test, real inputs_embeds, seq_len=913, bf16)

Test | PCC (approx.)
-- | --
12 layers + norm (real inputs) | 0.951
12 layers + norm + lm_head (real inputs) | 0.944 (matches whole model)
12 layers + norm (random same shape) | 0.999 (passes @ 0.99)

Layer depth vs PCC (real inputs_embeds, same sanity file)

Layers | PCC (approx.)
-- | --
1 | 0.9998
2 | 0.9958
4 | 0.9895
6 | 0.9839
8 | 0.9726
10 | 0.9647
12 | 0.9513

### Steps to reproduce the issue

```
git checkout akannan/deepseek_pcc_drop

# Isolated decoder-stack sanities
pytest tests/torch/models/deepseek_ocr/test_deepseek_v2_decoder_stack.py -svv 
```

### Expected behavior

Progressive / full DeepSeek OCR path PCC ≥ 0.99 vs CPU on doc.png inference inputs (decoder stack on real inputs_embeds, and whole-model logits).

**Comment by ashokkumarkannan1** [comment]
Classifying this as layer-wise PCC accumulation (isolated layers/slices pass; full depth fails). Same bucket as https://github.com/tenstorrent/tt-xla/issues/2413, https://github.com/tenstorrent/tt-xla/issues/4725, https://github.com/tenstorrent/tt-xla/issues/3786. Needs a shared tool/approach for accumulation-class drops, will revisit when that tooling is ready.
